Source/Orts.Simulation/Simulation/AIs/AI.cs

```diff
@@ -54,7 +54,18 @@ public class AI
         public List<AITrain> AutoGenTrains = new List<AITrain>(); // auto-generated trains
         public double clockTime; // clock time : local time before activity start, common time from simulator after start
         private bool localTime;  // if true : clockTime is local time
-        public bool PreUpdate; // if true : running in pre-update phase
+        public bool PreUpdate // if true : running in pre-update phase
+        {
+            get
+            {
+                return Simulator.PreUpdate;
+            }
+
+            set
```

Comment:
Giving read access to Simulator.PreUpdate is easy to understand.
But why is write access also needed?

Reply:
The PreUpdate variable is set from the AI class. Perhaps this should be protected, but the variable in Simulator is completely public anyway.

Reply:
Apparently, this is only set in one place, so how about we make that location set via `Simulator` and remove the setter here?

Reply:
It is set in 4 places: 3 in AI.cs and 1 in Simulator.cs.
Ok, I will change this since this is a low number of instances.

Reply:
Well... in the end, I removed it from AI and changed all of the calls to this variable.